### PR TITLE
Prefix x-amz-meta- for custom metadata

### PR DESCRIPTION
**Pull request by poornas**
if not specified.

Fixes: #353

### CODE REVIEW

File: Minio/ApiEndpoints/ObjectOperations.cs [hunk 1]
```diff
@@ -216,16 +218,22 @@ public async Task PutObjectAsync(string bucketName, string objectName, Stream da
...
             utils.ValidateBucketName(bucketName);
             utils.ValidateObjectName(objectName);
+
             var sseHeaders = new Dictionary<string, string>();
+            var meta = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            if (metaData != null) {
+                foreach (KeyValuePair<string, string> p in metaData)
+                {
+                    var key = p.Key;
+                    if (!supportedHeaders.Contains(p.Key) && !p.Key.ToLower().StartsWith("x-amz-meta-"))
```
Comment (Praveenrajmani) [inline Minio/ApiEndpoints/ObjectOperations.cs]: Can we write a separate function to check this? As we are using it in more than one place.

**Comment by Praveenrajmani** [comment]
This PR also fixes https://github.com/minio/minio-dotnet/issues/309

**Comment by aevitas** [comment]
Just an FYI, this change broke compatibility with DigitalOcean spaces for its canned ACLs.

**Comment by harshavardhana** [comment]
@aead please provide a reproducer and open a new issue

**Pull request merged**